runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java

```diff
@@ -105,7 +105,14 @@ public DataflowUnboundedReadFromBoundedSource(BoundedSource<T> source) {
 
   @Override
   public String getKindString() {
-    return "Read(" + approximateSimpleName(source.getClass()) + ")";
+    String sourceName;
+    if (source.getClass().isAnonymousClass()) {
+      sourceName = "AnonymousSource";
+    } else {
+      sourceName = approximateSimpleName(source.getClass());
+    }
+
+    return "Read(" + sourceName + ")";
```

Comment:
I could be wrong, but I think that (1) getKindString acts as the default name of a transform (2) changing names is not update-compatible. If both of these are correct, then changing anonymous sources to be named "Read(AnonymousSource)" would be update-incompatible, which we may want to avoid. Why is this change necessary for this PR?